Framework/CurveFitting/src/Algorithms/Fit.cpp

```diff
@@ -212,8 +209,16 @@ void Fit::finalizeMinimizer(size_t nIterations) {
                    std::to_string(m_maxIterations) + " iterations.";
   }
 
+  if (errorString.empty()) {
+    errorString = "success";
+  }
+
   // return the status flag
   setPropertyValue("OutputStatus", errorString);
+  setProperty("OutputNIterations", static_cast<int>(nIterations));
+  if (!this->isChild()) {
+    g_log.notice(errorString);
```

Comment:
To be honest I am not sure which level to use here but I don't want it to be `error` for now. Some of the unsuccessful results may be still acceptable to the user and I wouldn't want to annoy them by saying it's an error especially if we couldn't offer a better solution that returns `success`.